chap12/12-1.rb

```diff
@@ -1 +1,99 @@
-# 
\ No newline at end of file
+# 以下に示す自動車クラスCarに対して、総走行距離を表すフィールドと、その値を調べるメソッドを追加した自動車クラスExCarを作成せよ。クラスCarから派生すること。
+require 'date'
+class Car
+    attr_reader :name, :width, :height, :length, :x, :y, :fuel, :purchase_day
+    def initialize(name, width, height, length, x, y, fuel, purchase_day)
+        @name = name
+        @width = width
+        @height = height
+        @length = length
+        @x = x
+        @y = y
+        @fuel = fuel
+        @purchase_day = purchase_day
+    end
+
+    def get_x
+        x
+    end
+
+    def get_y
+        y
+    end
+
+    def remaining_fuel
+        fuel
+    end
+
+    def show_spec
+        puts "名前 : #{name}"
+        puts "車幅 : #{width} mm"
+        puts "車高 : #{height} mm"
+        puts "車長 : #{length} mm"
+    end
+
+    def move(dx, dy)
+        # 移動距離を算出
+        distance = Math.sqrt((dx ** 2) + (dy ** 2))
+
+        # distance走るだけのガソリンが残っているのか評価。燃費は1とみなす。
+        if distance > fuel
+            puts '燃料が足りません。'
+            return
+        else # ガソリンが足りるなら以下のような処理を。
+            @x += dx
+            @y += dy
+            @fuel -= distance
+        end
+    end
+end
+
+class ExCar < Car
+    def initialize(name, width, height, length, x, y, fuel, purchase_day)
+        @name = name
+        @width = width
+        @height = height
+        @length = length
+        @x = x
+        @y = y
+        @fuel = fuel
+        @purchase_day = purchase_day
+        @total_mileage = 0.0
+    end
```

Comment:
こんな感じに親のコンストラクタを明示的に呼び出して単純にできます。
```suggestion
  def initialize(name, width, height, length, x, y, fuel, purchase_day)
     super
    @total_mileage = 0.0
  end
```